### PR TITLE
Anton/contribute delete job

### PR DESCRIPTION
**Pull request by antondelchev**
Added an option to modify version (choosing synchronous or asynchronous processing) for the delete_job method. Related to #91

### CODE REVIEW

File: src/zos_jobs/zowe/zos_jobs_for_zowe_sdk/jobs.py [hunk 1]
```diff
@@ -89,17 +89,26 @@ def delete_job(self, jobname, jobid):
...
+        if modify_version not in ("1.0", "2.0"):
+            raise ValueError('Accepted values: "1.0" or "2.0".')
+
         custom_args = self._create_custom_request_arguments()
         job_url = "{}/{}".format(jobname, jobid)
         request_url = "{}{}".format(self.request_endpoint, job_url)
         custom_args["url"] = request_url
-        response_json = self.request_handler.perform_request("DELETE", custom_args, expected_code=[202])
+        custom_args["headers"]["X-IBM-Job-Modify-Version"] = "2.0"
+        if modify_version == "1.0":
```
Comment (boris-bc) [inline src/zos_jobs/zowe/zos_jobs_for_zowe_sdk/jobs.py:108]: This is not needed if you directly assign the value of `modify_version` to it.

File: src/zos_jobs/zowe/zos_jobs_for_zowe_sdk/jobs.py [hunk 2]
```diff
@@ -89,17 +89,26 @@ def delete_job(self, jobname, jobid):
...
         """
+        if modify_version not in ("1.0", "2.0"):
+            raise ValueError('Accepted values: "1.0" or "2.0".')
+
         custom_args = self._create_custom_request_arguments()
         job_url = "{}/{}".format(jobname, jobid)
         request_url = "{}{}".format(self.request_endpoint, job_url)
         custom_args["url"] = request_url
-        response_json = self.request_handler.perform_request("DELETE", custom_args, expected_code=[202])
+        custom_args["headers"]["X-IBM-Job-Modify-Version"] = "2.0"
```
Comment (boris-bc) [inline src/zos_jobs/zowe/zos_jobs_for_zowe_sdk/jobs.py:107]: You already verify on line 100 that `modify_version` is `1.0` or `2.0`. You can just assign the value directly to the custom argument.

File: tests/unit/test_zos_jobs.py [hunk 3]
```diff
@@ -1,15 +1,15 @@
...
 
-class TestJobsClass(unittest.TestCase):
+class TestJobsClass(TestCase):
     """Jobs class unit tests."""
 
     def setUp(self):
         """Setup fixtures for Jobs class."""
-        self.connection_dict = {"host": "https://mock-url.com",
+        self.test_profile = {"host": "https://mock-url.com",
                                 "user": "Username",
```
[the remark this answers is not in the document]
Comment (boris-bc) [reply tests/unit/test_zos_jobs.py:13]: This is still not resolved.

**Review by boris-bc — CHANGES_REQUESTED** [review]

**Review by boris-bc — CHANGES_REQUESTED** [review]

File: src/zos_jobs/zowe/zos_jobs_for_zowe_sdk/jobs.py (hunk 2 above)
Comment (antondelchev) [reply src/zos_jobs/zowe/zos_jobs_for_zowe_sdk/jobs.py]: Assigned the value of modify_version to custom_args["headers"]["X-IBM-Job-Modify-Version"]

File: src/zos_jobs/zowe/zos_jobs_for_zowe_sdk/jobs.py (hunk 1 above)
Comment (antondelchev) [reply src/zos_jobs/zowe/zos_jobs_for_zowe_sdk/jobs.py]: Assigned the value of modify_version to custom_args["headers"]["X-IBM-Job-Modify-Version"]

File: tests/unit/test_zos_jobs.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (antondelchev) [reply tests/unit/test_zos_jobs.py]: Fixed the indentation.

File: tests/unit/test_zos_jobs.py [hunk 4]
```diff
@@ -18,5 +18,13 @@ def setUp(self):
...
     def test_object_should_be_instance_of_class(self):
         """Created object should be instance of Jobs class."""
-        jobs = Jobs(self.connection_dict)
-        self.assertIsInstance(jobs, Jobs)
\ No newline at end of file
+        jobs = Jobs(self.test_profile)
+        self.assertIsInstance(jobs, Jobs)
+
+    @mock.patch("requests.Session.send")
+    def test_delete_job(self, mock_send_request):
```
[the remark this answers is not in the document]
Comment (antondelchev) [reply tests/unit/test_zos_jobs.py:26]: Added parameterized test to the method.

**Pull request merged**